src/functions/utils.ts

```diff
@@ -0,0 +1,29 @@
+import path from 'path';
+import fs from 'fs-extra';
+import https, { AgentOptions } from 'https';
+import Config from './config';
+
+export function setHttpsAgent(config: Config): https.Agent | undefined {
+  const { agentOptions } = config;
+  let options: AgentOptions | undefined;
+
+  if (agentOptions) {
+    options = (['ca', 'cert', 'key'] as const).reduce<AgentOptions>(
+      (acc, key) => {
+        if (typeof agentOptions[key] === 'string') {
+          return {
+            ...acc,
+            [key]: fs.readFileSync(path.resolve(agentOptions[key] as string)),
```

Comment:
odd that typescript doesnt recognize this as a string (because you check on it :P )

Reply:
We still using a pretty old version of typescript think that's why.